p2pserver/message/utils/msg_handler.go

```diff
@@ -676,3 +683,14 @@ func saveRespCache(key string, value interface{}) bool {
 	respCache.Add(key, value)
 	return true
 }
+
+func checkDuplicateTx(tx *types.Transaction) bool {
```

Comment:
If the function name is checkDupTx, it should be readonly, and not add tx hash to cache.
Otherwise, suggest changing function name to addTxHash?